src/main/java/eventprocessing/responseservices/ResponseProcessor.java

```diff
@@ -1,24 +1,43 @@
-package eventprocessing.responseservices;
-
-import eventprocessing.models.Response;
-import eventprocessing.models.SensorList;
-import eventprocessing.storage.MessageLog;
-
-
-public class ResponseProcessor {
-
-    public boolean isWorkingSensor(Response response, SensorList sensorList) {
-        long count = sensorList.getSensors().stream()
-                .filter(sensor -> sensor.getId().equals(response.getMessage().getLocationId()))
-                .count();
-        return count > 0;
-    }
-
-    public boolean isDuplicateMessage(Response response, MessageLog messageLog) {
-        return messageLog.getMessageHistory().contains(response.getMessageId());
-    }
-
-    public boolean isValidMessage(Response response, SensorList sensorList, MessageLog messageLog) {
-        return isWorkingSensor(response, sensorList) && !isDuplicateMessage(response, messageLog);
-    }
-}
+package eventprocessing.responseservices;
+
+import eventprocessing.models.BucketManager;
+import eventprocessing.models.SqsResponse;
+import eventprocessing.models.Bucket;
+import eventprocessing.models.SensorList;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class ResponseProcessor {
+    private final static Logger logger = LogManager.getLogger("ResponseProcessor");
+
+    public boolean isWorkingSensor(SqsResponse sqsResponse, SensorList sensorList) {
+        long count = sensorList.getSensors()
+                .stream()
+                .filter(sensor -> sensor.getId().equals(sqsResponse.getMessage().getLocationId()))
+                .count();
+        return count > 0;
+    }
+
+    public boolean isValidMessage(SqsResponse sqsResponse, SensorList sensorList, BucketManager bucketManager) {
+        boolean isValid = true;
+        for(Bucket bucket : bucketManager.getBuckets()) {
+            if(!isWorkingSensor(sqsResponse, sensorList)) {
+                isValid = false;
+            }
+            if (isDuplicateMessage(sqsResponse, bucket)) {
+                isValid = false;
+            }
+        }
+        return isValid;
+    }
+
+    public boolean isValidMessage(SqsResponse sqsResponse, SensorList sensorList, Bucket bucket) {
```

Comment:
It's a bit suspicious that you have two different methods `isValidMessage` which are used in subtley differerent ways - one taking a `Bucket` and one taking a `BucketManager`. Hopefully pushing more behaviour onto `BucketManager` will help you clean this up.